crates/starknet-devnet-core/src/messaging/mod.rs

```diff
@@ -61,6 +61,9 @@ pub struct MessagingBroker {
     pub l2_to_l1_messages_hashes: HashMap<String, u64>,
     /// This list of messages that will be sent to L1 node at the next `postman/flush`.
     pub l2_to_l1_messages_to_flush: Vec<MessageToL1>,
+    /// Mapping of L1 transaction hash to a chronological sequence of generated L2 transactions.
+    /// Using array of bytes as key to prevent format confusion with converting hash to hex.
+    pub l1_to_l2_tx_hashes: HashMap<[u8; 32], Vec<Felt>>,
```

Comment:
didnt understand why the key is bytes array

Reply:
E.g. `"0x1"` and `"0x01"` are the same number, but don't generate the same hash. That's what I meant by "format confusion". I can expand the comment if you agree.

Reply:
Also, forgot to mention, the original type of the L1 value hash (prior to converting to byte array) is not hashable.

Reply:
why it can't be H256 type

Reply:
True, it can be, I guess I didn't want to convert to another type (Hash256 -> H256)

Reply:
I know also what motivated me, in the code, just above the `l1_to_l2_tx_hashes` specification, there is the description of `l2_to_l1_messages_hashes` saying:
```rust
    /// Note:
    /// `Hash256` is not directly supported as a HashMap key due to missing trait.
    /// Using `String` instead.
```

So I wanted to be smarter than that and use bytes instead of String.